sushy_oem_idrac/utils.py

```diff
@@ -38,3 +43,32 @@ def reboot_system(system):
         system.refresh()
 
     LOG.info('System powered ON')
+
+@retrying.retry(
+    retry_on_exception=lambda exception: isinstance(exception, Exception),
+    stop_max_attempt_number=_IDRAC_IS_READY_RETRIES,
+    wait_fixed=_IDRAC_IS_READY_RETRY_DELAY_SEC * 1000)
+def wait_for_idrac_ready(oem_manager):
+    is_ready_response = is_idrac_ready(oem_manager)
+    if "LCStatus" in is_ready_response:
```

Comment:
The above check isn't sufficient to determine if the iDRAC is ready.  Here's the data when it's ready:
data={u'RTStatus': u'NotReady', u'Status': u'Ready', u'@Message.ExtendedInfo': [{u'Severity': u'OK', u'RelatedProperties': [], u'MessageId': u'Base.1.5.Success', u'MessageArgs@odata.count': 0, u'RelatedProperties@odata.count': 0, u'Message': u'Successfully Completed Request', u'Resolution': u'None', u'MessageArgs': []}, {u'Severity': u'Informational', u'RelatedProperties': [], u'MessageId': u'IDRAC.2.1.LC061', u'MessageArgs@odata.count': 0, u'RelatedProperties@odata.count': 0, u'Message': u'Lifecycle controller is ready to take provisioning requests', u'Resolution': u'No response action is required.', u'MessageArgs': []}], u'**LCStatus**': u'**Ready**', u'ServerStatus': u'PoweredOff'}

Here's the data when it's not:
data={u'RTStatus': u'NotReady', u'Status': u'NotReady', u'@Message.ExtendedInfo': [{u'Severity': u'OK', u'RelatedProperties': [], u'MessageId': u'Base.1.5.Success', u'MessageArgs@odata.count': 0, u'RelatedProperties@odata.count': 0, u'Message': u'Successfully Completed Request', u'Resolution': u'None', u'MessageArgs': []}, {u'Severity': u'Warning', u'RelatedProperties': [], u'MessageId': u'IDRAC.2.1.LC060', u'MessageArgs@odata.count': 0, u'RelatedProperties@odata.count': 0, u'Message': u'Lifecycle controller is currently not ready to take provisioning requests', u'Resolution': u'Wait until Lifecycle controller finishes performing current tasks and host system to finish POST.', u'MessageArgs': []}], u'**LCStatus**': u'**InUse**', u'ServerStatus': u'LCUSC'}

You can see that the string "LCStatus" is in the response, but the iDRAC is not ready.  We need to get the value of LCStatus and check to see if it's "Ready"

To put the iDRAC into the non-ready state, boot the server into the lifecycle controller.  When it is at that menu, it is not ready.

Reply:
Addressed.
Added the logic to detect "idrac is ready" as python-dracclient does after reset completion.
Now added check for 'LCStatus' value.